### ISSUE TITLE
APB unit tests: Random failures due to informer/lister cache not updated after event in queue

### ISSUE THREAD

**Issue by jordigilh** [ovn-org/ovn-kubernetes#3896]
In some rare cases, particularly when the host that runs the unit tests is slightly stressed (>50% cpu consumption), it is possible that the informer cache might return the previous instance of an object instead of the latest one, despite the fact that the event queue used by the APB controller has been updated with the latest object from the informer itself. In essence, this can lead to a situation where upon execution of the on<Add|Update|Delete>() function, the underlying call to the lister to retrieve the latest version of the given object, which is in the queue, returns the previous one instead, which leads to the test to fail.
 
The workaround at this point is to rerun the failed job for now.

**Comment by jordigilh** [comment]
FYI @npinaeva @trozet

**Comment by npinaeva** [comment]
When I debugged this, I figured the failures (at least the ones I saw) are caused by the status update, which on a test server causes the full object update. So if e.g. test code updated the APBRoute and then the status was updated, the whole object will be updated to the previous version. I have a WIP https://github.com/ovn-org/ovn-kubernetes/pull/3750 to fix this, maybe we should continue this investigation after we fix this part
